gtsam/nonlinear/Marginals.cpp

```diff
@@ -28,15 +28,35 @@ namespace gtsam {
 /* ************************************************************************* */
 Marginals::Marginals(const NonlinearFactorGraph& graph, const Values& solution, Factorization factorization,
                      EliminateableFactorGraph<GaussianFactorGraph>::OptionalOrdering ordering)
-{
+                     : values_(solution), factorization_(factorization) {
   gttic(MarginalsConstructor);
-
-  // Linearize graph
   graph_ = *graph.linearize(solution);
+  computeBayesTree(ordering);
+}
+
+/* ************************************************************************* */
+Marginals::Marginals(const GaussianFactorGraph& graph, const VectorValues& solution, Factorization factorization,
+                     EliminateableFactorGraph<GaussianFactorGraph>::OptionalOrdering ordering)
```

Comment:
Let's just use `boost::optional<Ordering>` everywhere? Including in .h file?

Reply:
done